zenpy/lib/request.py

```diff
@@ -219,6 +220,15 @@ def post(self, fp, token=None, target_name=None, content_type=None):
             else:
                 target_name = target_name or fp.name
 
+        if hasattr(fp, 'name'):
+            # Path objects (pathlib.Path)
+            if fp.name == '':
+                raise ZenpyException("upload requires a target file name")
+
+            target_name = target_name or fp.name
+
+            fp = open(fp, 'rb')
```

Comment:
Just realized we have a file leak (existing issue) as we don't close the file. Could you close this file object after the post?